src/test/ui/lint/lint-level-macro-def.rs

```diff
@@ -0,0 +1,17 @@
+// Checks that you can set a lint level specficially for a macro definition.
+//
+// This is a regression test for issue #59306.
+//
+// check-pass
+
+
+#[deny(missing_docs)]
+mod module {
+    #[allow(missing_docs)]
+    #[macro_export]
+    macro_rules! hello {
+        () => ()
+    }
+}
```

Comment:
Could you add a test case for this case
```rust
#[allow(missing_docs)]
mod module {
    #[macro_export]
    macro_rules! hello {
        () => ()
    }
}
```
, it'll make sure that macros exported from non-root modules still have those modules as parents from linting point of view.